[#1] [part-10] Add CI configuration

MrDmitry opened the pull request:
Migrate to OpenAL submodule
Fix windows build

MrDmitry committed:
[#1] Add doxygen configuration file

MrDmitry committed:
[#1] Add CI configurations

Migrate to OpenAL submodule

ilia-glushchenko commented on demos/basic/main.cpp:
```diff
@@ -33,12 +33,12 @@
 }
 
-constexpr double tau()
+double tau()
```
Would it be useful to have both versions available?

MrDmitry replied:
GCC and clang supply `constexpr std::atan` whereas CL doesn't. Using `constexpr variable` instead.

ilia-glushchenko commented on demos/basic/main.cpp:
```diff
@@ -33,12 +33,12 @@
 static std::string const testFile(TULPAR_DEMO_WORKING_DIR"data/ding_02.ogg");
 
-constexpr double pi()
+double pi()
```
Would it be useful to have both versions available?

MrDmitry replied:
Ditto

ilia-glushchenko commented on TulparAudio/internal/source/SourceCollection.cpp:
```diff
@@ -321,7 +321,16 @@ audio::Buffer SourceCollection::GetSourceActiveBuffer(SourceHandle source) const
+            }
+            else
+            {
+                buffer = queue.at(0);
```
Is this queue guaranteed to be at least of size 1 or otherwise should user code handle a possible exception?

MrDmitry replied:
Ultimately type of current source is provided by OpenAL. To avoid relying on some internal mechanisms I would rather check for `queue.empty()` prior to requesting the index.

MrDmitry committed:
[#15] Configure CMake packaging

Install OpenAL headers as part of tulpar_extra
Statically link OpenAL library

MrDmitry committed:
[#15] Fix release build issues

Extend travis build confirations with Release build

MrDmitry committed:
[#15] Use modern CMake features

MrDmitry committed:
[#1] Fix OpenAL PIC issue

MrDmitry committed:
Merge branch 'task_3/10/ci' into task_3/11/packaging

MrDmitry committed:
Merge branch 'task_3/11/packaging' into task_3/12/modern_cmake

MrDmitry committed:
[WIP] Fix clang debug test failure

MrDmitry committed:
[#1] Fix OpenAL linkage on Windows

MrDmitry committed:
[#1] Use OBJECT targets instead of STATIC internal libraries

This partially reverts commit 548b65761761be4af95c8d7ac57133a9975024ce

At the moment MSVC is unable to export all symbols from static libraries
As a workaround OBJECT targets are used

MrDmitry committed:
[#15] Use modern CMake features for demo

MrDmitry committed:
[#15] Use modern CMake features for tests

MrDmitry committed:
[#15] Fix symbols export on Windows platform

MrDmitry committed:
Merge pull request #17 from Godlike/task_3/12/modern_cmake

[#15] [part-12] Use modern CMake features

ilia-glushchenko commented:
approve

MrDmitry committed:
Merge pull request #16 from Godlike/task_3/11/packaging

[#15] [part-11] Configure CMake packaging
